{{cookiecutter.project_slug}}/clients/vue3/src/store/index.js

```diff
@@ -1,21 +1,44 @@
 import { createStore } from 'vuex'
 import createPersistedState from 'vuex-persistedstate'
+import { SET_USER } from './mutation-types'
 
 const STORAGE_HASH = 'Ax4XDbqIDE'
 export const STORAGE_KEY = `{{ cookiecutter.project_name }}-${STORAGE_HASH}`
 
-const state = {}
-const mutations = {}
-const actions = {}
-const modules = {}
-const getters = {}
+const state = {
+  user: null,
+}
+
+const mutations = {
+  [SET_USER]: (state, payload) => {
+    state.user = payload
+  },
+}
+
+const actions = {
+  setUser({ commit }, user) {
+    commit(SET_USER, user)
+  },
+}
+
+const getters = {
+  isLoggedIn: (state) => {
+    return !!state.user
+  },
+  user: (state) => {
+    return state.user
+  },
+  token: (state) => {
+    return state.user ? state.user.token : null
+  },
```

Comment:
You also need to consider the case where an auth-protected page is accessed from an external link or page refresh. There might be a user in the store, but you should validate their API token and redirect them before showing the page.

Here's some VueJS 2 Festi code that might be a helpful reference: https://github.com/thinknimble/festi/blob/main/webapp/src/store/index.js

Reply:
I'm going to go ahead and approve and merge this PR.
Yes there are some improvement that should be made, but login/signup are currently broken and this PR fixes that.
(I'll also be using the bootstrapper today so this will be a good live test)

Reply:
Will open an issue for the comments made here.